config/config.go

```diff
@@ -399,8 +394,33 @@ func (c KubeClusterSettings) K8sNetworkPlugin() string {
 func (c Cluster) Config() (*Config, error) {
 	config := Config{Cluster: c}
 
-	config.MinControllerCount = config.ControllerCount - 1
-	config.MaxControllerCount = config.ControllerCount + 1
+	// setup controller ASG size, one is the default ControllerCount
+	if config.ControllerCount != 1 && (config.ControllerASG.MinSize != 0 || config.ControllerASG.MaxSize != 0) {
+		return nil, fmt.Errorf("`controllerASG.MinSize` and `controllerASG.MaxSize` can only be specified without `controllerCount`")
+	}
+	if config.ControllerASG.MinSize == 0 {
+		config.ControllerASG.MinSize = config.ControllerCount
+	}
+	if config.ControllerASG.MaxSize == 0 {
+		config.ControllerASG.MaxSize = util.IntMax(config.ControllerCount, config.ControllerASG.MinSize)
+	}
+	if config.ControllerASG.RollingUpdateMinInstancesInService == 0 {
+		config.ControllerASG.RollingUpdateMinInstancesInService = config.ControllerASG.MaxSize - 1
+	}
+
+	// setup worker ASG size, one is the default WorkerCount
+	if config.WorkerCount != 1 && (config.WorkerASG.MinSize != 0 || config.WorkerASG.MaxSize != 0) {
+		return nil, fmt.Errorf("`workerASG.MinSize` and `workerASG.MaxSize` can only be specified without `workerCount`")
+	}
+	if config.WorkerASG.MinSize == 0 {
+		config.WorkerASG.MinSize = config.WorkerCount
+	}
+	if config.WorkerASG.MaxSize == 0 {
+		config.WorkerASG.MaxSize = util.IntMax(config.WorkerCount, config.WorkerASG.MinSize)
+	}
+	if config.WorkerASG.RollingUpdateMinInstancesInService == 0 {
+		config.WorkerASG.RollingUpdateMinInstancesInService = config.WorkerASG.MaxSize - 1
+	}
```

Comment:
Only static, default values set in `NewDefaultCluster()` are inherited to node pool config.
IMHO currently there's no timing to inherit "computed" values like these.
That is one of reasons I believe it is better to implement "computed" values by wrapping them into functions, instead of mutating values in place. ref https://github.com/coreos/kube-aws/pull/142/files#r91641540